typing_tester.py

```diff
@@ -0,0 +1,173 @@
+import curses
+from curses import wrapper
+import time
+import requests
+
+def get_random_word():
+    """Get random word with pronunciation from RapidApi"""
+
+    url = "https://random-words-api-umber.vercel.app/word"
+
+    headers = {
+        "X-RapidAPI-Key": "f934e406b8msh16f9dee78137dfcp1fab91jsn560d9925327d",
+        "X-RapidAPI-Host": "random-words-with-pronunciation.p.rapidapi.com"
+    }
+
+    response = requests.get(url, headers=headers)
+
+    word_obj = response.json()[0]
+
+    return word_obj
+
+
+def cal_accuracy_score(given_word, user_text, no_of_backspaces):
+    """Calculates the accuracy score for typing tester"""
+
+    wrong_letter = 0
+    try:
+        wrong_letter = sum([1 for i in range(len(given_word)) if user_text[i] != given_word[i]])
+    except:
+        wrong_letter += 1
+
+    total_mistakes = wrong_letter + no_of_backspaces
+    accuracy = len(given_word) - total_mistakes
+    # assign 0 if accuracy is negative
+    accuracy = 0 if accuracy < 0 else accuracy
+    accuracy =  accuracy/len(given_word)*100
+
+    return float(f"{accuracy:.2f}"), wrong_letter
+
+
+def cal_time_score(elapsed_time, given_word):
+    """Calculates the time score for typing tester"""
+
+    # Avg. time for one keystroke
+    keystroke_time = 0.28 
+    # Avg. time to hover hands from mouse to keyboard
+    mouse_to_keyboard_time = 0.4 
+    # formula to calculate raw score
+    raw_score = elapsed_time / (mouse_to_keyboard_time + (keystroke_time * (len(given_word) + 1)))
+
+    if raw_score < 1:
+         return float(100)
+    else:
+         return float(f"{(100/raw_score):.2f}")
+
+def display_result(stdscr, elapsed_time, word_obj, user_text, backspaces):
+      """Displays result of typing tester"""
+    
+      # Gets word, definition and pronunciation from word object
+      word = word_obj['word']
```

Comment:
Figure out what's the `.get` method of the dictionary does and why do we use it instead of using brackets

Reply:
Resolved